File: Backend/src/models/address_emails.py
# Backend/src/models/address_emails.py
# Definition of the AddressEmails model that connects addresses and emails.

# Author: Valentin Haas, 2024

# System imports
from typing import Optional

# Library imports
from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_pascal
from sqlalchemy import Column, Integer, Boolean
from sqlalchemy.ext.declarative import declarative_base

# Constants
BASE = declarative_base()


class AddressEmailsTable(BASE):
    """Definition of the AddressEmails model for the database."""

    __tablename__ = "AddressEmails"

    Address = Column(Integer, primary_key=True, foreign_key="Addresses.Id")
    """The unique identifier of the address."""

    Email = Column(Integer, primary_key=True, foreign_key="Emails.Id")
    """The unique identifier of the email."""

    IsPrimary = Column(Boolean, nullable=False, default=True)
    """Indicates whether the email is the primary email for the address."""


class AddressEmail(BaseModel):
    """Definition of the AddressEmail model for the API."""

    model_config = ConfigDict(populate_by_name=True, alias_generator=to_pascal)

    address: int
    """The unique identifier of the address."""

    email: int
    """The unique identifier of the email."""

    is_primary: Optional[bool] = True
    """Indicates whether the email is the primary email for the address."""

File: Backend/src/models/addressemails.go
// Backend/src/models/addressemails.go
// Definition of the AddressEmails model that connects addresses and emails.

// Author: Valentin Haas, 2025
package models

import (
	"gorm.io/gorm"
)

// AddressEmails represents the model that connects addresses and emails in the database.
type AddressEmail struct {
	gorm.Model // Provides ID, CreatedAt, UpdatedAt, DeletedAt fields

	Address   Address
	AddressID uint  `gorm:"primaryKey"` // The address of the email with a primary key constraint.
	Email     Email // Referenced Email Object
	EmailID   uint  `gorm:"primaryKey"`            // The email of the address with a primary key constraint.
	IsPrimary bool  `gorm:"default:true;not null"` // Indicates whether the email is the primary one for the address.
}
